File: eco_restoration_shard/cyboquatics/2026-08-17-bc-pfas-fog-corridor/lua/fog_router.lua
local function numeric_argument(index, name)
    local value = tonumber(arg[index])
    assert(value and value == value and value ~= math.huge and value ~= -math.huge, name .. " must be finite numeric")
    assert(value >= 0.0, name .. " must be non-negative")
    return value
end

local function classify(
    oil_mg_l,
    tss_mg_l,
    turbidity_ntu,
    oil_threshold_mg_l,
    tss_threshold_mg_l,
    turbidity_threshold_ntu
)
    assert(oil_threshold_mg_l > 0.0, "oil threshold must be positive")
    assert(tss_threshold_mg_l > 0.0, "TSS threshold must be positive")
    assert(turbidity_threshold_ntu > 0.0, "turbidity threshold must be positive")

    local fog = oil_mg_l > oil_threshold_mg_l
        and tss_mg_l > tss_threshold_mg_l
        and turbidity_ntu > turbidity_threshold_ntu

    if fog then
        return {
            p_fog = 1,
            media_state = "UNMODELED_MIXED_OIL_WATER_SEDIMENT",
            route = "MANUAL_FIELD_REVIEW",
            knowledge_factor = 0.45,
            eco_impact_value = 0.35,
            harm_risk = 0.75
        }
    end

    return {
        p_fog = 0,
        media_state = "FIELD_SCREENED_NOT_ALL_FOG_CONDITIONS_MET",
        route = "RETAIN_FOR_SAMPLING_AND_CONTEXT_REVIEW",
        knowledge_factor = 0.60,
        eco_impact_value = 0.50,
        harm_risk = 0.35
    }
end

if #arg ~= 6 then
    io.stderr:write(
        "usage: lua fog_router.lua <oil_mg_L> <TSS_mg_L> <turbidity_NTU> " ..
        "<oil_threshold_mg_L> <TSS_threshold_mg_L> <turbidity_threshold_NTU>\n"
    )
    os.exit(64)
end

local ok, result = pcall(
    classify,
    numeric_argument(1, "oil_mg_L"),
    numeric_argument(2, "TSS_mg_L"),
    numeric_argument(3, "turbidity_NTU"),
    numeric_argument(4, "oil_threshold_mg_L"),
    numeric_argument(5, "TSS_threshold_mg_L"),
    numeric_argument(6, "turbidity_threshold_NTU")
)

if not ok then
    io.stderr:write("error: " .. result .. "\n")
    os.exit(65)
end

print("P_fog=" .. result.p_fog)
print("media_state=" .. result.media_state)
print("route=" .. result.route)
print(string.format("knowledge_factor=%.2f", result.knowledge_factor))
print(string.format("eco_impact_value=%.2f", result.eco_impact_value))
print(string.format("harm_risk=%.2f", result.harm_risk))

File: eco_restoration_shard/cyboquatics/2026-08-17-bc-pfas-fog-corridor/kotlin/FogRouter.kt
data class FogThresholds(
    val oilMgL: Double,
    val tssMgL: Double,
    val turbidityNtu: Double
)

data class FogDecision(
    val pFog: Int,
    val mediaState: String,
    val route: String,
    val knowledgeFactor: Double,
    val ecoImpactValue: Double,
    val harmRisk: Double
)

fun requireFiniteNonNegative(value: Double, name: String) {
    require(value.isFinite() && value >= 0.0) { "$name must be finite and non-negative" }
}

fun fogRoute(
    oilMgL: Double,
    tssMgL: Double,
    turbidityNtu: Double,
    thresholds: FogThresholds
): FogDecision {
    requireFiniteNonNegative(oilMgL, "oilMgL")
    requireFiniteNonNegative(tssMgL, "tssMgL")
    requireFiniteNonNegative(turbidityNtu, "turbidityNtu")
    require(thresholds.oilMgL > 0.0 && thresholds.tssMgL > 0.0 && thresholds.turbidityNtu > 0.0) {
        "field-calibrated thresholds must be positive"
    }

    val pFog = oilMgL > thresholds.oilMgL &&
        tssMgL > thresholds.tssMgL &&
        turbidityNtu > thresholds.turbidityNtu

    return if (pFog) {
        FogDecision(
            pFog = 1,
            mediaState = "UNMODELED_MIXED_OIL_WATER_SEDIMENT",
            route = "MANUAL_FIELD_REVIEW",
            knowledgeFactor = 0.45,
            ecoImpactValue = 0.35,
            harmRisk = 0.75
        )
    } else {
        FogDecision(
            pFog = 0,
            mediaState = "FIELD_SCREENED_NOT_ALL_FOG_CONDITIONS_MET",
            route = "RETAIN_FOR_SAMPLING_AND_CONTEXT_REVIEW",
            knowledgeFactor = 0.60,
            ecoImpactValue = 0.50,
            harmRisk = 0.35
        )
    }
}

fun main(args: Array<String>) {
    require(args.size == 6) {
        "usage: FogRouterKt <oil_mg_L> <TSS_mg_L> <turbidity_NTU> " +
            "<oil_threshold_mg_L> <TSS_threshold_mg_L> <turbidity_threshold_NTU>"
    }

    val decision = fogRoute(
        oilMgL = args[0].toDouble(),
        tssMgL = args[1].toDouble(),
        turbidityNtu = args[2].toDouble(),
        thresholds = FogThresholds(
            oilMgL = args[3].toDouble(),
            tssMgL = args[4].toDouble(),
            turbidityNtu = args[5].toDouble()
        )
    )

    println("P_fog=${decision.pFog}")
    println("media_state=${decision.mediaState}")
    println("route=${decision.route}")
    println("knowledge_factor=%.2f".format(decision.knowledgeFactor))
    println("eco_impact_value=%.2f".format(decision.ecoImpactValue))
    println("harm_risk=%.2f".format(decision.harmRisk))
}
